### DISCUSSION TITLE
Please help: Oqtane and Interserver hosting

### DISCUSSION THREAD

**Discussion by Loclhh** [oqtane/oqtane.framework#2145] [Q&A] [answered]
Please tell me if anyone know how to hosting Oqtane to Interserver.net. I found it has a cheaper hosting price compared to microsoft. I search on google and found that it can support Asp.Net. How about if it can support Oqtane? If anyone has experiment with this hosting server, pls give me some instruction.

Many thanks!

Loc

**Comment by leigh-pointer** [comment]
@Loclhh I have been using  Interserver.net for over a year now and support is very good.  You just need to be aware of the webDAV issue.  WebDAV needs to be switched off in the web.Config file as it is not possible to do it through the web control panel.  Read [WebDav and disabling](https://www.studio-elf.net/mynews?Aid=35) on what needs to be applied to the web.config file.

Any other assistance you need, I am will to help you out as best I can.

**Comment by Loclhh** [comment]
Dear @leigh-pointer,

Some members in the support team of Interserver give me some instructions but it didn't help me solve the problem up to now. 
On April 21, based on a supporter's instruction, she send me the link:  http://up.is.cc/download.php?f=nodwmq_WebDeploy.pdf that has some guides for Web deployment to the server. Based on that, I did try publish my web project (C# .NET 6) directly from Visual Studio 2022 but got error: "Failed to connect to the remote computer" with details: "Could not complete the request to remote agent URL 'https://mydomain.com:8172/msdeploy.axd?site=mydomain.com'. The underlying connection was closed: An unexpected error occurred on a send."  Then I disable my firewall on my desktop and try again then I still got the same message. After that, I make new very simple web project and try to publish based on the supporter's instruction for testing but still had the same problem.

Yesterday, instead publishing directly from visual studio as I did before, I published my web project to a publish folder on my desktop then upload all the files in the folder to the httpdocs by using File Manager tools on their website then I can go to my website success. However, when I create database for my website with the page database configuration (oqtane database configuration) then it got error as shown in picture below:
![image](https://user-images.githubusercontent.com/52992727/164608204-9846f984-832a-4d52-b443-969fd6fbd4a3.png)

I would be very appreciated if you could help me some guide steps that I could follow to solve my problem.

I looking for hearing from you with many thanks.

Loc

**Comment by leigh-pointer** [comment]
Hi @Loclhh take a quick read here.  https://github.com/oqtane/oqtane.framework/discussions/1207

**Reply by Loclhh** [reply]
Dear @leigh-pointer ,

Which version 3.1.0, Webconfig already have tag <remove name="WebDAVModule" /> in <handlers> and <modules>. I believe that the problem was caused by something in configurations in Plesk. But I don't what it is. I had reported my problem to the support team and waiting for reply.

do you have any advise?

Many thanks.

**Comment by leigh-pointer** [comment, marked as answer]
@Loclhh 
I set up my database in plesk and then in the install page, pointed the database to the one created in plesk.  I also created a DBLogin for that database.  I dont think you can use Intergrated Security.

You will also need to add a SSL certificate to your domain.  This is done for free and automatically updated.

1. In Pleskl Goto Websites & Domains
2. Click on SSL/TLS Certificates
3. Create your certificates

For the rest I didnt really do much.  
I would also add, I always use the FTP deploy with the files from github.

**Reply by Loclhh** [reply]
Dear @leigh-pointer, 

I'm sorry for delay in reply your message because few days ago I disscussed with support team of InterServer by emails and they had been enabled SSL and 32 bit mode on their server. Now, I can publish my website directly from visual studio 2022 to their server susscessfully. However, I have another problem when create database from oqtane page (the initial page - database configuration page ) that when I click "Install Now" then the errror: "An error occurred. This application may no longer respond until reloaded." as picture below:
![image](https://user-images.githubusercontent.com/52992727/165009389-e09e649a-36a9-4da1-a2ae-db14372fba20.png)

I turn on stdoutLogEnabled="true" in web.config file to get information of the error on Plesk then I find out some info about the error:
Request: https://mydomain.com/api/Installation
Response status: Forbidden ModSecurity Action
warn: Microsoft.AspNetCore.Components.Server.Circuits.RemoteRenderer[100]
      Unhandled exception rendering component: Object reference not set to an instance of an object.
      System.NullReferenceException: Object reference not set to an instance of an object.
         at Oqtane.Installer.Installer.Install() in D:\Oqtane\3.1.0\sourcecode\Interserver\oqtane.framework-3.1.0\Oqtane.Client\Installer\Installer.razor:line 201
         at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
         at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
fail: Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost[111]
      Unhandled exception in circuit '5Yz52RhnaOeW6j4njXyLxMaBF4CRq80MSMVZaGoyDBk'.
      System.NullReferenceException: Object reference not set to an instance of an object.
         at Oqtane.Installer.Installer.Install() in D:\Oqtane\3.1.0\sourcecode\Interserver\oqtane.framework-3.1.0\Oqtane.Client\Installer\Installer.razor:line 201
         at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
         at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[1]
      Executed endpoint '/_blazor'
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished HTTP/1.1 GET https://mydomain.com/_blazor?id=05TLyZdG1V821fjtR4dNOw - - - 101 - - 1567750.3844ms
info: Microsoft.Asp

With the information I think that something wrong in permission on the website because the error information said:
Request: https://mydomain.com/api/Installation
Response status: Forbidden ModSecurity Action

Please be noted that I did set full right to the folder httpdocs on Plesk:
![image](https://user-images.githubusercontent.com/52992727/165009778-3691367f-42cc-4ccd-bed7-fa783c78173f.png)

I also did the trick that you advised that I create database directly from Plesk's database tool then migrate all information from my local database that I run successfully from local then change the "name" field value of table Alias to consist with connection information of my website on plesk. After finish this, I can go to main page of my website (main page of oqtane). However, I cannot login to my website with user host (the only one user I created for testing).  Everytime I login with Host the website said there is something wrong in login.

Do you have any idea?

Loc

**Reply by leigh-pointer** [reply]
Hi @Loclhh You said you were publishing from VisualStuido ?  This is not the right way to go.  You should run the release.cmd in the Packagers folder or use the Install package available on GitHub.

**Reply by Loclhh** [reply]
hi @leigh-pointer, I did try both ways: Install package on GitHub and from Visual Studio. Both have the same error: "An error occurred. This application may no longer respond until reloaded."

**Reply by Loclhh** [reply]
I also asked help from Interserver's support team but they refuse and said that they only assist technical problems not code problems :(

**Reply by Loclhh** [reply]
hi @leigh-pointer 
What is the version of Oqtane framwork you deploy to interserver? I guess it may be affect by some changing in latest version of oqtane.

Regards,

**Reply by leigh-pointer** [reply]
I have run Oqtane for the last year on interserver and I am currently running 3.1.0.

**Reply by leigh-pointer** [reply]
Installing from Visual Studio will not work as the Release.cmd package contains configurations that VS does not.

**Reply by Loclhh** [reply]
Thank you very much @leigh-pointer. 
I wil try install which previous version of oqtane from install package from Github. If it success with previous version in create database in "database configuration" then I can upgrade to the latest version. Hope this will be success.

**Reply by leigh-pointer** [reply]
Hi @Loclhh Remember your DB user Login needs to be in the login security Schema. This was a problem for me and support added to the schema because for some reason it was not automated.

**Reply by Loclhh** [reply]
Thank you very much @leigh-pointer, 

I will forward this information the the support team and hope this can solve my problem.

**Reply by Loclhh** [reply]
hi @leigh-pointer,

The support said that he enabled my DB user Login to the login security Schema but I still got the same error before. I think the problem at my db login permission.

Could you follow this thread (https://github.com/oqtane/oqtane.framework/discussions/2154) as I did report my problem to @sbwalker with hope that he could advise me how to solve it. 

If you have any advise please make a note in the thread.

Thank you very much!

**Reply by Loclhh** [reply]
Hi @leigh-pointer ,

I did deploy my website successfully. :)  

I also listed all steps I did for other people if they need it in thread: https://github.com/oqtane/oqtane.framework/discussions/2154

Thank you very much for your advice and I'm very appreciated with all your help.

Cheer, 

Loc
